codegen/type_converter_test.go

```diff
@@ -1102,28 +1102,27 @@ func TestConvertWithBadImportMapOfBadStruct(t *testing.T) {
 	)
 }
 
-func TestConvertWithBadKeyMapOfString(t *testing.T) {
+func TestConvertWithCorrectKeyMapNotStringKey(t *testing.T) {
 	lines, err := convertTypes(
 		"Foo", "Bar",
-		`struct Foo {
-			1: optional map<i32, string> one
+		`
+		typedef string UUID
+
+		struct Foo {
+			1: optional map<UUID, string> one
 			2: required map<i32, string> two
 		}
 
 		struct Bar {
-			1: optional map<i32, string> one
+			1: optional map<UUID, string> one
 			2: required map<i32, string> two
 		}`,
 		nil,
 		nil,
 	)
 
-	assert.Error(t, err)
-	assert.Equal(t, "", lines)
-	assert.Equal(t,
-		"could not convert key (one), map is not string-keyed.",
-		err.Error(),
-	)
+	assert.NoError(t, err)
+	assert.Equal(t, "out.One = make(map[structs.UUID]string, len(in.One))\nfor key, value := range in.One {\n\tout.One[ structs.UUID(key)] = string(value)\n}\nout.Two = make(map[int32]string, len(in.Two))\nfor key, value := range in.Two {\n\tout.Two[ int32(key)] = string(value)\n}", lines)
```

Comment:
for reviewability and sanity please use 

    trim(`
        // code with whitespace
    `)